From bc7e3566079cdcfe2eabe0e7331098051c074b0c Mon Sep 17 00:00:00 2001
From: Dirk-Jan Faber <dfaber@victronenergy.com>
Date: Mon, 3 Aug 2026 12:27:07 +0200
Subject: [PATCH 1/3] package.json: no more optional packages

1) change optional packages to mandatory
This changes nothing to the build, as they were already always installed. But
not anymore as per the recent change to npm-online-install.bbclass.

For the reasons why each package is wanted see here:
https://docs.google.com/document/d/1oJLH7MoRAPoqlo9aaO0gnBYA05a2ruQWFsexZfqWm9E

For why we don't want packages to be listed as optional, see here:
https://github.com/victronenergy/venus/wiki/npm-and-nodejs-packages

Part of https://github.com/victronenergy/venus/issues/1304

2) add signalk-venus-plugin because we need that

3) update @mxtommy/kip and @signalk/freeboard-sk

And do that such that its done automatically when refreshing the shrinkwrap
---
 package.json | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/package.json b/package.json
index de7e6b87..96d71b9a 100644
--- a/package.json
+++ b/package.json
@@ -135,9 +135,7 @@
     "unified": "^11.0.5",
     "unzipper": "^0.12.3",
     "uuid": "^8.3.2",
-    "ws": "^8.17.0"
-  },
-  "optionalDependencies": {
+    "ws": "^8.17.0",
     "@mxtommy/kip": "^4.8.0",
     "@signalk/app-dock": "^1.1.0",
     "@signalk/freeboard-sk": "^2.24.0",
@@ -148,7 +146,8 @@
     "@signalk/udp-nmea-plugin": "^2.0.0",
     "serialport": "^11.0.1",
     "signalk-n2kais-to-nmea0183": "^2.0.3",
-    "signalk-to-nmea2000": "^2.24.0"
+    "signalk-to-nmea2000": "^2.24.0",
+    "signalk-venus-plugin": "^2.4.1"
   },
   "devDependencies": {
     "@eslint-react/eslint-plugin": "^1.53.1",
-- 
2.43.0

